code/datums/components/riding.dm

```diff
@@ -414,3 +420,12 @@
 		var/mob/living/simple_animal/S = parent
 		S.toggle_ai(AI_ON)
 	..()
+
+/datum/component/riding/proc/on_emp_act(datum/source, severity)
+	SIGNAL_HANDLER
+
+	emped = TRUE
+	addtimer(CALLBACK(src, PROC_REF(reboot)), 300, TIMER_UNIQUE|TIMER_OVERRIDE) //if a new EMP happens, remove the old timer so it doesn't reactivate early
```

Comment:
Adjust the time according to the severity (Higher severity is less severe, so divide 300 by severity)